Added ESP component and SESP (stub ESP model).

gold2718 opened the pull request:
- Added an eighth component type, ESP to CIME.
- Added the stub ESP model, SESP.
- Made seq_flds_x2c_fluxes and seq_flds_c2x_fluxes inputs to component_run optional (ESP components do not communicate with coupler).
- Created new interfaces to seq_infodata_getData and seq_infodata_putData which take a component one-letter-id and make the correct retrievals or settings. Use new interfaces in component_mod.
- Use new parameters to clarify how many communicators are needed for a run.
- Use ID = -1 to indicate an idea which is not attached to a seq_comms object.

Test suite:  CIME standalone tests, FC5
Test baseline: cesm1_5_alpha06
Test namelist changes: NA
Test status: bit for bit

Fixes: NA

User interface changes?: Optional ESP component type in compset longname

Code review:  Pending

Commit:
Added ESP component and SESP (stub ESP model).

Added  an eighth component type, ESP to CIME.
Added the stub ESP model, SESP.
Made seq_flds_x2c_fluxes and seq_flds_c2x_fluxes inputs to
component_run optional (ESP components do not communicate with coupler).
Created new interfaces to seq_infodata_getData and seq_infodata_putData
which take a component one-letter-id and make the correct retrievals or
settings. Use new interfaces in component_mod.
Use new parameters to clarify how many communicators are needed for a run.
Use ID = -1 to indicate an idea which is not attached to a seq_comms object.

Test suite:  CIME standalone tests, FC5
Test baseline: cesm1_5_alpha06
Test namelist changes: NA
Test status: bit for bit

Fixes: NA

User interface changes?: Optional ESP component type in compset longname

Code review:  Pending

jedwards4b commented on cime_config/cesm/machines/buildlib.csm_share:
```diff
@@ -26,6 +26,7 @@ my $NINST_LND		= `./xmlquery NINST_LND		-value`;
 my $NINST_OCN		= `./xmlquery NINST_OCN		-value`;
 my $NINST_ROF		= `./xmlquery NINST_ROF		-value`;
 my $NINST_WAV		= `./xmlquery NINST_WAV		-value`;
+my $NINST_ESP		= `./xmlquery NINST_ESP		-value`;
```
Are we allowing for more than one instance of this component?

gold2718 replied:
I do not see any reason to restrict it. While the immediate goal is DART (an ensemble filter), future applications, such as geoengineering, may have the same number of instances as the atmosphere.

jedwards4b commented on doc/modelnl/env_mach_pes.html:
```diff
@@ -649,6 +649,82 @@
     </td>
     <td>env_mach_pes.xml</td>
   </tr>
+  <tr id="NTASKS_ESP_tr">
```
whats all the java script stuff about?

gold2718 replied:
I don't see anything different here. This is just how the file is set up to make it searchable and expandable. The page seems to work fine as is.

Commit:
Added comment in component_run for ESP model use

Added a comment to both versions of component_run explaining how
to use the newly-optional arguments (seq_flds_x2c_fluxes and
seq_flds_c2x_fluxes) to avoid coupler-based actions for external (ESP)
components.

Test suite: NA
Test baseline: NA
Test namelist changes: NA
Test status: [bit for bit]

Fixes: NA

User interface changes?: [No]

Code review: This change was recommended in the 2016-04-15 code review

gold2718 commented:
I added the comments as requested in the code review. @jedwards4b, can you merge now?

jedwards4b commented:
done

On Mon, Apr 18, 2016 at 5:59 PM, goldy notifications@github.com wrote:

> Assigned #418 https://github.com/CESM-Development/cime/pull/418 to
> @jedwards4b https://github.com/jedwards4b.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/CESM-Development/cime/pull/418#event-632146480

## 

Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO
